test/e2e/framework/service/jig.go

```diff
@@ -58,6 +60,33 @@ var NodePortRange = utilnet.PortRange{Base: 30000, Size: 2768}
 // It is copied from "k8s.io/kubernetes/pkg/registry/core/service/portallocator"
 var errAllocated = errors.New("provided port is already allocated")
 
+// staticPortRange implements port allocation model described here
+// https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/3668-reserved-service-nodeport-range
+type staticPortRange struct {
+	sync.Mutex
+	baseport      int
+	length        int
+	reservedPorts sets.Set[int]
```

Comment:
You should use `sets.Set[int32]` (and likewise use `int32` in your APIs below). That should get rid of a bunch of typecasting.